src/AmpImage/__tests__/ApmImage.test.js

```diff
@@ -0,0 +1,167 @@
+import React from 'react'
+import { render } from '@testing-library/react'
+import AmpImage from '..'
+import { image, imageWithPreferred } from '../../__testdata__/image'
+
+// This is basically a more limited selection of the same tests from Image, since the functionality is very similar, but the code is repeated, so they could change independently. I have added checks for those few AMP specific features, such as styles and needing a width and height.
+
+function objectMother() {
+  const srcSet =
+    'https://img.apmcdn.org/c2c452354fbff94d720ba8f86e2c71ba7427b306/uncropped/35bd3b-20181220-serena-brook-opens-our-show-at-the-town-hall.jpg 400w,https://img.apmcdn.org/c2c452354fbff94d720ba8f86e2c71ba7427b306/uncropped/f5db37-20181220-serena-brook-opens-our-show-at-the-town-hall.jpg 600w,https://img.apmcdn.org/c2c452354fbff94d720ba8f86e2c71ba7427b306/uncropped/04a63f-20181220-serena-brook-opens-our-show-at-the-town-hall.jpg 1000w,https://img.apmcdn.org/c2c452354fbff94d720ba8f86e2c71ba7427b306/uncropped/72bc48-20181220-serena-brook-opens-our-show-at-the-town-hall.jpg 1400w,https://img.apmcdn.org/c2c452354fbff94d720ba8f86e2c71ba7427b306/uncropped/f20034-20181220-serena-brook-opens-our-show-at-the-town-hall.jpg 2000'
+  const src =
+    'https://img.apmcdn.org/c2c452354fbff94d720ba8f86e2c71ba7427b306/uncropped/f5db37-20181220-serena-brook-opens-our-show-at-the-town-hall.jpg'
+  const sizes = '(min-width: 960px) 720px, 100vw'
+  const className = 'test'
+  const alt = 'Some nice lawn chairs'
+  const fallbackSrc =
+    'https://s3-us-west-2.amazonaws.com/s.cdpn.io/298/wolf_20131015_003_1400.jpg'
+  const fallbackSrcSet =
+    'https://s3-us-west-2.amazonaws.com/s.cdpn.io/298/wolf_20131015_003_1400.jpg 1400w, https://s3-us-west-2.amazonaws.com/s.cdpn.io/298/wolf_20131015_003_700.jpg 700w, https://s3-us-west-2.amazonaws.com/s.cdpn.io/298/wolf_20131015_003_400.jpg 400w'
+
+  return {
+    src,
+    srcSet,
+    sizes,
+    class: className,
+    alt,
+    fallbackSrc,
+    fallbackSrcSet
+  }
+}
+
+// SUCCESSES
+
+test('has styles required to make amp-img not expand off the page', () => {
+  const { container } = render(<AmpImage image={image} />)
+  const img = container.firstChild
+
+  expect(img).toHaveAttribute(
+    'style',
+    expect.stringContaining('max-width: 100%')
+  )
+})
+
+test('provides width and height required by AMP based on instances', () => {
```

Comment:
uses uncropped instance when no preferred aspect ration is present  as test name?